file/random_access_file_reader.cc

```diff
@@ -93,7 +102,17 @@ IOStatus RandomAccessFileReader::Read(
   IOStatus io_s;
   uint64_t elapsed = 0;
   {
-    StopWatch sw(clock_, stats_, hist_type_,
+    IOActivity io_activity_to_use =
+        op_io_activity != IOActivity::kUnknown ? op_io_activity : io_activity_;
+    StopWatch sw_sst_read_mircos(
+        clock_, stats_, SST_READ_MICROS,
+        (stats_ != nullptr && (io_activity_to_use == IOActivity::kCompaction ||
+                               io_activity_to_use == IOActivity::kFlush))
+            ? &elapsed
+            : nullptr,
+        true /*overwrite*/, true /*delay_enabled*/);
```

Comment:
Tried avoid nested StopWatch but even storing a list or something like below causes 4% regression in readrandom, not sure if something we should take.

```
      uint32_t time = (elapsed_ != nullptr)
                          ? *elapsed_
                          : (clock_->NowMicros() - start_time_);
      statistics_->reportTimeToHistogram(
          hist_type_, time);
      if (hist_type_ == Histograms::SST_READ_FLUSH_MICROS || hist_type_ == Histograms::SST_READ_COMPACTION_MICROS){
         statistics_->reportTimeToHistogram(
          SST_READ_MICROS, time);
      }
 ```
 
Downside of the current approach: SST_READ_MICROS = 1.5 * the actual SST_READ_MICROS (see Test plan).

Reply:
Fixed